binary/reflect.go

```diff
@@ -803,6 +820,24 @@ func writeReflectJSON(rv reflect.Value, rt reflect.Type, w io.Writer, n *int64,
 			WriteTo([]byte("}"), w, n, err)
 		}
 
+	case reflect.Map:
+		WriteTo([]byte("{"), w, n, err)
+		keys := rv.MapKeys()
```

Comment:
Do we need to order the keys?  So far this is the only non-determinism in tendermint/binary.